lib/kubeclient/common.rb

```diff
@@ -267,8 +261,23 @@ def patch_entity(entity_type, name, patch, namespace = nil)
       end
     end
 
-    def new_entity(hash, klass)
-      klass.new(hash)
+    def new_entity(hash, entity_type)
+      resource_class(entity_type).new(hash)
+    end
+
+    def resource_class(entity_type)
+      # TODO: test binding since there already seems to be a constant with that name
+      Kubeclient.const_get(entity_type)
+      rescue NameError
+        Kubeclient.const_set(
```

Comment:
> Can we check if the entity actually exists?

When a request is made for a resource that does not exist we raise a [name error](https://github.com/abonas/kubeclient/pull/181/files#diff-104b3d1973988933ffc83feada470da2R71).
That is done per your previous suggestion, to be backward compatible.